app/basketball/[competition]/teams/page.tsx

```diff
@@ -0,0 +1,45 @@
+import Custom404 from "@/components/404";
+import SeasonDropDownMenu from "@/components/basketball/teams/TeamDropDowmMenu";
+import TeamList from "@/components/basketball/teams/TeamList";
+import { asyncFetch } from "@/utils/fetch";
+import { getRecentSeasonByGroup } from "@/utils/get-recent-seasons";
+import { GROUPNAME_TO_COMPETITIONID } from "@/utils/variables";
+
+export default async function Page({ params, searchParams }: any) {
+  // fetch current season
+  const season = await getRecentSeasonByGroup(params.competition);
```

Comment:
the ordering of this request is introducing useless requests
supppose to be sth like
```
if(season exists in queryParam){
fetch season by id
} else {
fetch reason season
}